### PR TITLE
feat: Implement member update

### LINKED ISSUE #6: Create CRUD endpoints for org members
As described on the README in principle, but as always feedback is most welcomed.

Probably best to create one PR per endpoint.

Skip auth enforcement if #2 isn't implemented.

### PR DESCRIPTION
**Pull request by levansuper**
Fixes #6

### CODE REVIEW

File: src/member.ts [hunk 1]
```diff
@@ -102,3 +102,38 @@ export async function deleteMember(
...
+  }
+
+  const role = memberData.role && ROLE_MAPPING[memberData.role];
+
+  const memberModel = getModelForClass(MemberModelSchema, {
+    existingConnection: options.dbConnection,
+  });
+
+  try {
+    await memberModel.findByIdAndUpdate(memberId, { ...memberData, role });
```
Comment (gnarea) [inline src/member.ts:123]: Does this work when `memberData.role` isn't set?

File: src/member.ts [hunk 2]
```diff
@@ -102,3 +102,38 @@ export async function deleteMember(
...
+  memberId: string,
+  memberData: PatchMemberSchema,
+  options: ServiceOptions,
+): Promise<Result<undefined, MemberProblemType>> {
+  const validationFailure = validateMemberData(memberData, options);
+  if (validationFailure !== undefined) {
+    return { didSucceed: false, reason: validationFailure };
+  }
+
+  const role = memberData.role && ROLE_MAPPING[memberData.role];
```
Comment (gnarea) [inline src/member.ts:116]:
```suggestion
  const role = memberData.role ?? ROLE_MAPPING[memberData.role];
```

That way, `role` type becomes `MemberRole | null`. (If I'm not mistaken, right now it's `MemberRole | false`)

File: src/member.spec.ts [hunk 3]
```diff
@@ -261,4 +261,166 @@ describe('member', () => {
       expect(error).toHaveProperty('name', 'MongoNotConnectedError');
     });
   });
+
+  describe('updateMember', () => {
+    const testEmail = 'different@email.com';
+
+    test('Valid should be updated', async () => {
```
Comment (gnarea) [inline src/member.spec.ts:268]: This is testing the scenario where all the fields are set, but I think the opposite (nothing is changed) is also useful.

File: src/services/routes/member.routes.spec.ts [hunk 4]
```diff
@@ -273,4 +277,135 @@ describe('member routes', () => {
...
+
+      const response = await serverInstance.inject({
+        ...injectionOptions,
+        payload: {},
+      });
+
+      expect(response).toHaveProperty('statusCode', HTTP_STATUS_CODES.NO_CONTENT);
+    });
+
+    test.each(memberSchemaRoles)('Role %s should be allowed', async (role: MemberSchemaRole) => {
```
Comment (gnarea) [inline src/services/routes/member.routes.spec.ts:311]: I'm not sure this test is useful.

**Review by gnarea — CHANGES_REQUESTED** [review]

File: src/services/routes/member.routes.spec.ts (hunk 4 above)
Comment (levansuper) [reply src/services/routes/member.routes.spec.ts:311]: This tests the role enum in the schema

**Review by gnarea — CHANGES_REQUESTED** [review]

File: src/models/Member.model.ts [hunk 5]
```diff
@@ -7,13 +7,17 @@ export enum Role {
...
   { orgName: 1, name: 1 },
-  { unique: true, partialFilterExpression: { name: { $exists: true } } },
+  { unique: true, partialFilterExpression: { name: { $type: 'string' } } },
 )
 export class MemberModelSchema {
-  @prop()
+  @prop({
+    default: null,
+  })
   public name?: string;
```
Comment (gnarea) [inline src/models/Member.model.ts:16]: This type is `string | undefined`, which is wrong. It should be `string | null`:

```suggestion
  public name!: string | null;
```

**Review by gnarea — CHANGES_REQUESTED** [review]

**Review by gnarea — APPROVED** [review]
Thanks!

**Pull request merged**